platforms/jvm/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/ApiGroovyCompiler.java

```diff
@@ -372,13 +374,9 @@ private VersionNumber parseGroovyVersion() {
     // This is necessary because:
     // 1. serialization/deserialization of the compile spec doesn't preserve Boolean.TRUE/Boolean.FALSE but creates new instances
     // 1. org.codehaus.groovy.classgen.asm.WriterController makes identity comparisons
-    @SuppressWarnings("ModifyCollectionInEnhancedForLoop")
-    private void canonicalizeValues(Map<String, Boolean> options) {
-        for (String key : options.keySet()) {
-            // unboxing and boxing does the trick
-            boolean value = options.get(key);
-            options.put(key, value);
-        }
+    private static void canonicalizeValues(Map<String, Boolean> options) {
```

Comment:
```suggestion
    private static void promotePrimitiveBooleanToClass(Map<String, Boolean> options) {
```

does this reflect the actual doing? As `canonicalize` is quite generic, lacking the details considered on this abstraction lvl.
On short scope, long names fit well.